src/components/fields/DataSelector.js

```diff
@@ -94,6 +95,14 @@ export class UnconnectedDataSelector extends Component {
       fromSrc: this.context.srcConverters ? this.context.srcConverters.fromSrc : null,
     });
 
+    if (this.props.container.type) {
```

Comment:
well, actually, it's possible not to provide trace.type in data, then plotly.js defaults to scatter..
maybe we should encode this in the editor.. this will only be potentially a problem for figures created with an api, where the person did not specify trace.type

Reply:
oh actually nevermind, I think we'd be handling that here: https://github.com/plotly/react-chart-editor/blob/master/src/lib/customTraceType.js#L9